> @richardleach Please check if this now works for you. I followed @hvds advice
> 

When I try building this branch on FreeBSD-12 (my customary threaded build, `clang10`), `make` very quickly fails:
```
[perlmonger: perl] $ gitcurr
pr3
[perlmonger: perl] $ git describe
v5.37.1-83-g3d480457fa
[perlmonger: perl] $ make
echo @`sh  cflags "optimize='-O2 -pipe -fstack-protector -fno-strict-aliasing'" opmini.o`  -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
@cc -c -DPERL_CORE -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -std=c99 -O2 -pipe -fstack-protector -fno-strict-aliasing -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings -DPERL_IS_MINIPERL -DPERL_EXTERNAL_GLOB opmini.c
In file included from op.c:163:
In file included from ./perl.h:7494:
./sv_inline.h:456:10: error: invalid application of 'typeof' to bit-field
         ASSUME(type_details->arena);
         ^
./perl.h:4227:36: note: expanded from macro 'ASSUME'
        ( (void) ({     __typeof__ (R) R__assume_ = (R);          \
                                   ^
1 error generated.
*** Error code 1

Stop.
make: stopped in /usr/home/jkeenan/gitwork/perl
```

--- a/perl.h
+++ b/perl.h
@@ -4217,6 +4217,14 @@ intrinsic function, see its documents for more details.
 
 #ifdef DEBUGGING
 #  define ASSUME(x) assert(x)
+#elif __has_builtin(__builtin_assume)
+#  if defined(__clang__) || defined(__clang)
+#    define ASSUME(x)  CLANG_DIAG_IGNORE(-Wassume)      \
+                       __builtin_assume (x)             \
+                       CLANG_DIAG_RESTORE
+#  else
+#    define ASSUME(x)  __builtin_assume(x)
+#  endif
 #elif defined(_MSC_VER)
 #  define ASSUME(x) __assume(x)
 #elif defined(__ARMCC_VERSION) /* untested */

